### PR TITLE
feat(routes): add fastapi routes 

### PR DESCRIPTION
**Pull request by wxiwnd**
- get resource
- get resource detail
- get helper

### CODE REVIEW

File: kagami/routes/root.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                "data": {
+                    "title": announcement.title,
+                    "content": announcement.content,
+                    "date": announcement.date,
+                },
+            }
+            for announcement in announcements
+        ]
+    else:
+        raise HTTPException(status_code=404, detail="No announcement found")
```
Comment (sparrowhe) [inline kagami/routes/root.py]: It should be a 500 error code. "No announcement" can be an expected behavior, response with a empty array, but a internal (or database) error is not.

File: kagami/database/database_service/announce.py [hunk 2]
```diff
@@ -42,3 +43,6 @@ async def update_announce(
         if new_title:
             origin_announce.title = new_title
         await self._session.commit()
+
+    async def list_announce(self) -> list[Announce]:
+        return await self._session.execute(select(Announce))
```
Comment (sparrowhe) [inline kagami/database/database_service/announce.py]: Is there any limit on the amount of quried data? Consider add a paging to this.

**Review by sparrowhe — CHANGES_REQUESTED** [review]

File: kagami/routes/root.py (hunk 1 above)
Comment (wxiwnd) [reply kagami/routes/root.py]: That is true. WIP

File: kagami/database/database_service/announce.py (hunk 2 above)
Comment (wxiwnd) [reply kagami/database/database_service/announce.py]: That will have some changes on api definition. cc @GEORGE-Ta

Comment (wxiwnd) [reply kagami/database/database_service/announce.py]: I am thinking about adding page number and page size in api definition.

Comment (wxiwnd) [reply kagami/database/database_service/announce.py]: [2e95610](https://github.com/ProjectSoftoid/Kagami/pull/26/commits/2e9561006aa2298864a7bcee87d4fabc6e8b43bf)

**Comment by wxiwnd** [comment]
PTAL @sparrowhe @hedgehog098

**Review by sparrowhe — COMMENTED** [review]
Another suggestions is return the total page number to the frontend to render the pageing component. ~~If the frontend uses infinite scrolling to display announcements, this is not necessary.~~ Obviously, this is not appropriate.

**Comment by wxiwnd** [comment]
> Another suggestions is return the total page number to the frontend to render the pageing component. ~If the frontend uses infinite scrolling to display announcements, this is not necessary.~ Obviously, this is not appropriate.

Let's move on and talk about this problem in next PR.

**Comment by sparrowhe** [comment]
> > Another suggestions is return the total page number to the frontend to render the pageing component. ~If the frontend uses infinite scrolling to display announcements, this is not necessary.~ Obviously, this is not appropriate.
> 
> Let's move on and talk about this problem in next PR.

LGTM for other parts

**Pull request merged**